app/src/main/java/com/example/moviesapp/activity/MainActivity.kt

```diff
@@ -20,40 +18,28 @@ class MainActivity : AppCompatActivity(), KoinComponent {
 
     private val viewModel: MainActivityViewModel by inject()
     private lateinit var binding: ActivityMainBinding
-    private lateinit var pageAdapter: PageAdapter
-    private var isConnected = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        isConnected = Connection.isOnline(this)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-
-        binding.viewPager.adapter = PageAdapter(this)
-        pageAdapter = binding.viewPager.adapter as PageAdapter
-        TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
-            tab.text = pageAdapter.getPageTitle(position)
-        }.attach()
-
-        viewModel.getMovies(isConnected)
-        viewModel.getMovieState().observe({ lifecycle }, ::updateUI)
+        viewModel.getMainActivityState().observe({ lifecycle }, ::updateUI)
+        viewModel.getState()
```

Comment:
the name of this method should be initialize or something like that.. a get should response something and here is no response

Reply:
it's true, fixed